docs/release.md

```diff
@@ -0,0 +1,34 @@
+# Release
+This document describes how to release the ClusterAPI BYOH infrastructure provider.
```

Comment:
s/ClusterAPI/Cluster API/g
```suggestion
This document describes how to release the Cluster API BYOH infrastructure provider.
```

There are more places in the doc where we can make similar edits.